### PR TITLE
WIP - Fix infinite loop when retrying to get a connection

### LINKED ISSUE #440: Fix infinite loop when retrying to get a connection

### PR DESCRIPTION
**Pull request by nickkkccc**
I haven't forgotten about:
- [ ] Tests
- [ ] Changelog
- [ ] Documentation
- [ ] Commit messages comply with the [guideline](https://www.tarantool.io/en/doc/latest/dev_guide/developer_guidelines/#how-to-write-a-commit-message)
- [ ] Cleanup the code for review. See [checklist](https://github.com/tarantool/cartridge-java/blob/master/docs/review-checklist.md)

Related issues:
Closes #440

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +14,7 @@
 - Add `"mode"` option for select operation ([#107](https://github.com/tarantool/cartridge-java/issues/107))
 - Change using of proxy client parameters (`mode`, `rollback_on_error`, `stop_on_error`) with enum classes ([#419](https://github.com/tarantool/cartridge-java/issues/419))
 - Change using of option names with the String type in the internal API to using with the ProxyOption enum class ([#420](https://github.com/tarantool/cartridge-java/issues/420))
+- Bump `testcontainers-java-tarantool` to 1.0.2 ([#440](https://github.com/tarantool/cartridge-java/issues/440))
```
Comment (bitgorbovsky) [inline CHANGELOG.md:17]: Please, remove it

File: src/main/java/io/tarantool/driver/api/connection/TarantoolConnectionSelectionStrategies.java [hunk 2]
```diff
@@ -38,22 +38,29 @@ static final class RoundRobinStrategy implements ConnectionSelectionStrategy {
...
         private final TarantoolConnectionIterator connectionIterator;
         private final AtomicInteger available;
+        private final int connectionsCount;
 
         RoundRobinStrategy(Collection<TarantoolConnection> connections) {
-            this.available = new AtomicInteger(connections.size());
-            this.connectionIterator = new TarantoolConnectionIterator(connections.stream()
-                .peek(conn -> conn.addConnectionCloseListener(c -> available.getAndDecrement()))
-                .collect(Collectors.toList()));
+            this.connectionsCount = connections.size();
```
Comment (bitgorbovsky) [inline src/main/java/io/tarantool/driver/api/connection/TarantoolConnectionSelectionStrategies.java:44]: Please leave formatting as before

File: src/main/java/io/tarantool/driver/api/connection/TarantoolConnectionSelectionStrategies.java [hunk 3]
```diff
@@ -96,22 +105,24 @@ private Collection<TarantoolConnectionIterator> populateIterators(
             int groupSize = config.getConnections();
             AtomicInteger currentSize = new AtomicInteger(0);
             return connections.stream()
-                .peek(conn -> conn.addConnectionCloseListener(c -> available.getAndDecrement()))
-                .collect(Collectors.groupingBy(
-                    conn -> currentSize.getAndIncrement() / groupSize)).values().stream()
-                .map(TarantoolConnectionIterator::new)
-                .filter(TarantoolConnectionIterator::hasNext)
-                .collect(Collectors.toList());
+                              .peek(conn -> conn.addConnectionCloseListener(c -> available.getAndDecrement()))
```
Comment (bitgorbovsky) [inline src/main/java/io/tarantool/driver/api/connection/TarantoolConnectionSelectionStrategies.java:108]: Please leave formatting as before

**Review by bitgorbovsky — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (nickkkccc) [reply CHANGELOG.md:17]: Исправил

File: src/main/java/io/tarantool/driver/api/connection/TarantoolConnectionSelectionStrategies.java (hunk 2 above)
Comment (nickkkccc) [reply src/main/java/io/tarantool/driver/api/connection/TarantoolConnectionSelectionStrategies.java:44]: Исправил

File: src/main/java/io/tarantool/driver/api/connection/TarantoolConnectionSelectionStrategies.java (hunk 3 above)
Comment (nickkkccc) [reply src/main/java/io/tarantool/driver/api/connection/TarantoolConnectionSelectionStrategies.java]: Исправил

**Pull request merged**